### PR TITLE
Issue #2852300: Save currency code with unit cost.

### PR DESCRIPTION
**Pull request by steveoliver**
See https://www.drupal.org/node/2852300

### CODE REVIEW

**Comment by BBGuy** [comment]
- Adds 'unit_cost_currency_code' to schema - spot on 
- Looks like doesn't allow for zero cost as handles those as null - don't know if an issue
- needs fixing a merge conflict probably as a result of https://github.com/BBGuy/commerce_stock/commit/da7652d521a40a8478099c3609f11429118f464f

File: modules/local_storage/src/LocalStockUpdater.php [hunk 1]
```diff
@@ -58,21 +59,25 @@ public static function create(ContainerInterface $container) {
...
+  public function createTransaction(PurchasableEntityInterface $entity, $location_id, $zone, $quantity, Price $unit_cost, $transaction_type_id, array $metadata) {
     // Get optional fields.
     $related_tid = isset($metadata['related_tid']) ? $metadata['related_tid'] : NULL;
     $related_oid = isset($metadata['related_oid']) ? $metadata['related_oid'] : NULL;
     $related_uid = isset($metadata['related_uid']) ? $metadata['related_uid'] : NULL;
     $data = isset($metadata['data']) ? $metadata['data'] : NULL;
 
+    $cost_number = $unit_cost->isZero() ? NULL : $unit_cost->getNumber();
+    $cost_code = $unit_cost->isZero() ? NULL : $unit_cost->getCurrencyCode();
+
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockUpdater.php:71]: remove this for now -- save price even if amount is zero and currency code is XXX.

**Comment by steveoliver** [comment]
Rebased this.

**Comment by olafkarsten** [comment]
this one is obsolete. We did this somewhere.

**Pull request closed without merging**